envoy/upstream/outlier_detection.h

```diff
@@ -47,6 +50,63 @@ enum class Result {
   ExtOriginRequestSuccess // Request was completed successfully.
 };
 
+class HttpCode {
+public:
+  HttpCode(uint32_t code) : code_(code) {}
+  HttpCode() = delete;
+  uint32_t code() const { return code_; }
+
+private:
+  uint32_t code_;
+};
+
+// LocalOriginEvent is used to report errors like resets, timeouts but also
+// successful connection attempts.
+class LocalOriginEvent {
+public:
+  LocalOriginEvent(Result result) : result_(result) {}
+  LocalOriginEvent() = delete;
+  Result result() const { return result_; }
+
+private:
+  Result result_;
+};
+
+/*
+ * Class carries result of a transaction with upstream entity
+ * or generated internally by Envoy.
+ */
+using ExtResult = absl::variant<HttpCode, LocalOriginEvent>;
```

Comment:
Is there a possible the one caller want to report a result with both `Result` and status code? I see a this type calling:

```
    upstream_request.upstreamHost()->outlierDetector().putResult(result, code);
```

Reply:
No. It was done for situations when result has to be converted to Http code. This does not happen in the new extensions. In other words, local origin errors are never converted to http code. If a user wants to "catch" local origin events, it should define a new consecutive errors monitor which only "catch" local origin errors.